tusclient/request.py

```diff
@@ -22,6 +22,7 @@ def __init__(self, uploader):
         self.handle = pycurl.Curl()
         self.response_headers = {}
         self.output = six.StringIO()
+        self._status_code = None
```

Comment:
since we no longer have the handle after the request is closed, it may make sense to have just `self.status_code` as the attribute, while we delete the `@property` that we have now.